### PR TITLE
Add edit comments on AJAX

### PR DESCRIPTION
**Pull request by vakyl9r**

### CODE REVIEW

**Comment by vakyl9r** [comment]
@Magons take a look, pls)

**Comment by Magons** [comment]
@vakyl9r pls do rebase

File: app/assets/javascripts/comments.coffee [hunk 1]
```diff
@@ -1,21 +1,58 @@
...
+      success: (data) ->
+        $button.parent().find('.body_edit').hide()
+        $button.parent().find('.comment_body').show()
+        document.getElementById("comment_body_#{$comment_id}").innerHTML = $updated_comment_body
+        $button.parent().find('.edit_comment').show()
+        $button.parent().find('.delete_comment').show()
+        $button.parent().find('.cancel_update').remove()
+        $button.remove()
+      error: (data) ->
+        alert('Произошла непредвиденная ошибка.')
```
Comment (Magons) [inline app/assets/javascripts/comments.coffee:57]: here you can know about error. Call me about this

File: app/assets/javascripts/comments.coffee [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
+        <div class='comment_body inline' id='comment_body_#{data.id}'>#{data.body}</div>
+        <input class='comment_id' hidden value='#{data.id}'>
         <button class='btn btn-primary mt-10 delete_comment'>Удалить</button>
         <button class='btn btn-primary mt-10 edit_comment'>Редактировать</button>
         </a>")
         $('#comment_body').val('')
       error: (data) ->
         alert('Нельзя отправлять пустой комментарий. Пожалуйста, введите текст в поле комментария.')
+  $(document).on 'click', '.edit_comment', ->
+    $(this).parent().find('.comment_body').hide()
```
Comment (Magons) [inline app/assets/javascripts/comments.coffee]: `$(this).parent()` very common, create a variable for it

File: app/assets/javascripts/comments.coffee [hunk 3]
```diff
@@ -1,21 +1,58 @@
...
+    $(this).parent().append("<button class='btn update_comment'>Update</button>
+    <button class='btn cancel_update'>Cancel</button>")
+    $(this).parent().find('.delete_comment').hide()
+    $(this).hide()
+  $(document).on 'click', '.cancel_update', ->
+      $(this).parent().find('.comment_body').show()
+      $(this).parent().find('.body_edit').hide()
+      $(this).parent().find('.edit_comment').show()
+      $(this).parent().find('.delete_comment').show()
+      $(this).parent().find('.update_comment').remove()
```
Comment (Magons) [inline app/assets/javascripts/comments.coffee]: this common should be a separate function

**Comment by vakyl9r** [comment]
@Magons rebased

File: app/assets/javascripts/comments.coffee (hunk 1 above)
Comment (vakyl9r) [reply app/assets/javascripts/comments.coffee:57]: @Magons when can we call you?

Comment (Magons) [reply app/assets/javascripts/comments.coffee:57]: @vakyl9r now

**Comment by vakyl9r** [comment]
@Magons Fixed

File: app/assets/javascripts/comments.coffee [hunk 4]
```diff
@@ -1,21 +1,60 @@
...
+  $(document).on 'click', '.cancel_update', ->
+    $parent = $(this).parent()
+    hide_and_show($parent)
+    $parent.find('.update_comment').remove()
+    $(this).remove()
+  $(document).on 'click', '.update_comment', ->
+    $parent = $(this).parent()
+    $button = $(this)
+    post_id = $('#post_id').val()
+    $comment_id = $(this).parent().find('.comment_id').val()
```
Comment (Magons) [inline app/assets/javascripts/comments.coffee]: use $parent

File: app/assets/javascripts/comments.coffee [hunk 5]
```diff
@@ -1,21 +1,60 @@
...
+    $updated_comment_body = $(this).parent().find('.body_edit').val()
```
Comment (Magons) [inline app/assets/javascripts/comments.coffee]: use $parent

File: app/assets/javascripts/comments.coffee [hunk 6]
```diff
@@ -1,21 +1,60 @@
...
+    $.ajax
+      type: 'PUT'
+      url: "/posts/#{post_id}/comments/#{$comment_id}"
+      data: { comment: { body: $updated_comment_body } }
+      dataType: "json"
+      success: (data) ->
+        hide_and_show($parent)
+        $("#comment_body_#{$comment_id}")[0].innerHTML = $updated_comment_body
```
Comment (Magons) [inline app/assets/javascripts/comments.coffee:53]: 👍

**Pull request merged**